### DISCUSSION TITLE
Menu Wrapper, Unable to use <Menu.Target />

### DISCUSSION THREAD

**Discussion by shareef99** [mantinedev/mantine#3852] [Q&A] [answered]
Hello,
Thanks for making such a great library.
I got a question/doubt 

I am trying to make a menu wrapper and use it on different components.
My Basic Implementation

```
import { Menu as MMenu } from "@mantine/core";
import type { MenuProps } from "@mantine/core";

export default function Menu(props: MenuProps) {
  return <MMenu {...props}>{props.children}</MMenu>;
}
```

I am not able to use `<Menu.Target />` and other things menu exports.

Is there a context that I need to wrap menu, like

```
<MenuContext>
    <Menu />
</MenuContext>
```

Thanks in advance

**Comment by rtivital** [comment, marked as answer]
- There is no MenuContext component
- If you want to use static components, you need to assign them to your wrapper:

```ts
YourMenu.Target = Menu.Target;
// ...other components
```

**Reply by disillusionedowl** [reply]
Thanks @rtivital , I was also scratching my head on this. There is not much info on this topic even when i Google `react static component`.

The assignment code you have suggested seems like hard-coded boilerplate compared to classic function wrappers, higher-order functions, etc that should "just work" as one would expect. Must be a react thing.
